### PR TITLE
Add SourceFileProcessor, ValueConsumer

### PR DESCRIPTION
**Pull request by eric-swanson**
Add [SourceFileProcessor](https://github.com/polserver/polserver/blob/ecompile-antlr/pol-core/bscript/compiler/astbuilder/SourceFileProcessor.cpp) stub: visits a SourceFile, as well as any included .em or .inc files
Add ValueConsumer: An AST node that consumes the result of the evaluation of its child nodes
  - this is used for expression statements and for `var` statements

### CODE REVIEW

File: pol-core/bscript/compiler/astbuilder/SourceFileProcessor.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#ifndef POLSERVER_SOURCEFILEPROCESSOR_H
+#define POLSERVER_SOURCEFILEPROCESSOR_H
+
+#include <EscriptGrammar/EscriptParserBaseVisitor.h>
+
+#include <boost/optional.hpp>
+#include <memory>
+#include <string>
+
+#include "ProgramBuilder.h"
```
Comment (frozenblit) [inline pol-core/bscript/compiler/astbuilder/SourceFileProcessor.h:10]: If you want to keep the "no file from the same folder unless .h/.cpp pair"
```suggestion
#include "compiler/astbuilder/ProgramBuilder.h"
```

File: pol-core/bscript/compiler/astbuilder/SourceFileProcessor.cpp [hunk 2]
```diff
@@ -0,0 +1,38 @@
+#include "SourceFileProcessor.h"
+
+#include "../clib/fileutil.h"
+
+#include "BuilderWorkspace.h"
```
Comment (frozenblit) [inline pol-core/bscript/compiler/astbuilder/SourceFileProcessor.cpp:5]: If you want to keep the "no file from the same folder unless .h/.cpp pair"
```suggestion
#include "compiler/astbuilder/BuilderWorkspace.h"
```

**Review by frozenblit — APPROVED** [review]
Two include paths could still be changed but otherwise looks good to me

File: pol-core/bscript/compiler/astbuilder/SourceFileProcessor.cpp (hunk 2 above)
Comment (eric-swanson) [reply pol-core/bscript/compiler/astbuilder/SourceFileProcessor.cpp:5]: I see I missed a few others

File: pol-core/bscript/compiler/ast/ValueConsumer.cpp [hunk 3]
```diff
@@ -0,0 +1,25 @@
+#include "ValueConsumer.h"
+
+#include <format/format.h>
+
+#include "NodeVisitor.h"
```
Comment (eric-swanson) [inline pol-core/bscript/compiler/ast/ValueConsumer.cpp:5]:
```suggestion
#include "compiler/ast/NodeVisitor.h"
```

File: pol-core/bscript/compiler/ast/ValueConsumer.h [hunk 4]
```diff
@@ -0,0 +1,22 @@
+#ifndef POLSERVER_VALUECONSUMER_H
+#define POLSERVER_VALUECONSUMER_H
+
+#include "Statement.h"
```
Comment (eric-swanson) [inline pol-core/bscript/compiler/ast/ValueConsumer.h:4]:
```suggestion
#include "compiler/ast/Statement.h"
```

**Review by frozenblit — APPROVED** [review]

**Pull request merged**